apptools/persistence/tests/test_state_function.py

```diff
@@ -89,47 +82,25 @@ def test_normal_setstate(self):
         there are no registered state functions in the class chain.
         """
         # Validate that unpickling the first class gives us an instance of
-        # the third class with the appropriate attribute values.  It will have
+        # the second class with the appropriate attribute values.  It will have
         # the default Foo values (because there is no state function to move
-        # them) and also the default Baz values (since they inherit the
+        # them) and also the default Bar values (since they inherit the
         # trait defaults because nothing overwrote the values.)
         start = Foo()
-        end = sweet_pickle.loads(sweet_pickle.dumps(start))
-        self.assertIsInstance(end, Baz)
+        test_file = io.BytesIO(pickle.dumps(start, 2))
+        end = VersionedUnpickler(test_file, updater=TestUpdater()).load()
+        self.assertIsInstance(end, Bar)
         self._assertAttributes(end, 1, (False, 1, 1, "foo"))
-        self._assertAttributes(end, 2, None)
-        self._assertAttributes(end, 3, (False, 3, 3, "baz"))
+        self._assertAttributes(end, 2, (True, 2, 2, "bar"))
+        self._assertAttributes(end, 3, None)
 
         # Validate that unpickling the second class gives us an instance of
         # the third class with the appropriate attribute values.  It will have
         # only the Baz attributes with the Bar values (since the __setstate__
         # on Baz converted the Bar attributes to Baz attributes.)
         start = Bar()
-        end = sweet_pickle.loads(sweet_pickle.dumps(start))
-        self.assertIsInstance(end, Baz)
-        self._assertAttributes(end, 2, None)
-        self._assertAttributes(end, 3, (True, 2, 2, "bar"))
-
-    def test_unpickled_chain_functionality(self):
-        """Validates that the registered state functions are used when
-        unpickling.
-        """
```

Comment:
I believe it may be possible to modify this test, to get a similar flavor of test for `apptools.persistence` by using the `setstates` attribute on an `Updater` instance that is passed to a `VersionedUnpickler`.  Unfortunately I was unable to get it to work atm.

Reply:
I looked into this briefly... IMO, it looks evil! I opened https://github.com/enthought/apptools/issues/238.